test/sagemaker/test_elastic_inference.py

```diff
@@ -0,0 +1,114 @@
+    if region not in EI_SUPPORTED_REGIONS:
+        pytest.skip('EI is not supported in {}'.format(region))
+
+
+@pytest.fixture
+def pretrained_model_data(region):
+    return 's3://sagemaker-sample-data-{}/tensorflow/model/resnet/resnet_50_v2_fp32_NCHW.tar.gz'.format(region)
+
+
+def _timestamp():
+    return time.strftime("%Y-%m-%d-%H-%M-%S")
+
+
+def _execution_role(session):
+    return session.resource('iam').Role('SageMakerRole').arn
+
+
+def _production_variants(model_name, instance_type, accelerator_type):
+    production_variants = [{
+        'VariantName': 'AllTraffic',
+        'ModelName': model_name,
+        'InitialInstanceCount': 1,
+        'InstanceType': instance_type,
+        'AcceleratorType': accelerator_type
+    }]
+    return production_variants
+
+
+@pytest.mark.skip_if_non_supported_ei_region
+@pytest.mark.skip_if_no_accelerator
+def test_deploy_elastic_inference_with_pretrained_model(pretrained_model_data,
+                                                        docker_image_uri,
+                                                        instance_type,
+                                                        accelerator_type):
+    endpoint_name = 'test-tfs-ei-deploy-model-{}'.format(_timestamp())
+    endpoint_config_name = 'test-tfs-endpoint-config-{}'.format(_timestamp())
+    model_name = 'test-tfs-ei-model-{}'.format(_timestamp())
+
+    session = boto3.Session()
+    client = session.client('sagemaker')
+    runtime_client = session.client('runtime.sagemaker')
+    client.create_model(ModelName=model_name,
+                        ExecutionRoleArn=_execution_role(session),
+                        PrimaryContainer={
+                            'Image': docker_image_uri,
+                            'ModelDataUrl': pretrained_model_data
+                        })
+
+    logger.info('deploying model to endpoint: {}'.format(endpoint_name))
+
+    client.create_endpoint_config(EndpointConfigName=endpoint_config_name,
+                                  ProductionVariants=_production_variants(model_name, instance_type, accelerator_type))
+
+    client.create_endpoint(EndpointName=endpoint_name,
+                           EndpointConfigName=endpoint_config_name)
+
+    try:
+        client.get_waiter('endpoint_in_service').wait(EndpointName=endpoint_name)
+    finally:
+        status = client.describe_endpoint(EndpointName=endpoint_name)['EndpointStatus']
+        if status != 'InService':
+            raise Exception('Failed to create endpoint.')
+
+    input_data = {'instances': np.random.rand(1, 1, 3, 3).tolist()}
+
+    response = runtime_client.invoke_endpoint(EndpointName=endpoint_name,
+                                              ContentType='application/json',
+                                              Body=json.dumps(input_data))
+    result = json.loads(response['Body'].read().decode())
+    assert result['predictions'] is not None
+
+    client.delete_endpoint(EndpointName=endpoint_name)
```

Comment:
delete_endpoint should clear endpoint_config too
also needs to be in a `finally` block so delete happens even if test fails